packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -86,129 +101,70 @@ export abstract class VirtualGatewayListener {
+  constructor(private readonly protocol: Protocol,
+    private readonly healthCheck: HealthCheck | undefined,
+    private readonly port: number = 8080,
+    private readonly tlsCertificate: TlsCertificate | undefined) {
     super();
-    this.port = options.port ? options.port : 8080;
-    this.healthCheck = options.healthCheck;
   }
 
   /**
    * Called when the GatewayListener type is initialized. Can be used to enforce
    * mutual exclusivity
    */
-  public bind(_scope: cdk.Construct): VirtualGatewayListenerConfig {
+  public bind(scope: cdk.Construct): VirtualGatewayListenerConfig {
+    const tlsConfig = this.tlsCertificate?.bind(scope);
     return {
       listener: {
         portMapping: {
           port: this.port,
           protocol: this.protocol,
         },
-        healthCheck: this.healthCheck ? renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
+        healthCheck: this.healthCheck ? this.renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
+        tls: this.renderTls(tlsConfig),
       },
     };
   }
-}
-
-/**
-* Represents the properties needed to define an HTTP2 Listener for a VirtualGateway
-*/
-class Http2GatewayListener extends HttpGatewayListener {
-  constructor(options: HttpGatewayListenerOptions = {}) {
-    super(options);
-    this.protocol = Protocol.HTTP2;
-  }
-}
-
-/**
- * Represents the properties needed to define a GRPC Listener for Virtual Gateway
- */
-class GrpcGatewayListener extends VirtualGatewayListener {
-  /**
-   * Port to listen for connections on
-   *
-   * @default - 8080
-   */
-  readonly port: number;
 
   /**
-   * Health checking strategy upstream nodes should use when communicating with the listener
-   *
-   * @default - no healthcheck
+   * Renders the TLS config for a listener
    */
-  readonly healthCheck?: HealthCheck;
-
-  /**
-   * Protocol the listener implements
-   */
-  protected protocol: Protocol = Protocol.GRPC;
-
-  constructor(options: HttpGatewayListenerOptions = {}) {
-    super();
-    this.port = options.port ? options.port : 8080;
-    this.healthCheck = options.healthCheck;
-  }
+  private renderTls(tlsCertificateConfig: TlsCertificateConfig | undefined): CfnVirtualGateway.VirtualGatewayListenerTlsProperty | undefined {
+    if (tlsCertificateConfig === undefined) { return undefined; }
 
-  /**
-   * Called when the GatewayListener type is initialized. Can be used to enforce
-   * mutual exclusivity
-   */
-  public bind(_scope: cdk.Construct): VirtualGatewayListenerConfig {
     return {
-      listener: {
-        portMapping: {
-          port: this.port,
-          protocol: Protocol.GRPC,
-        },
-        healthCheck: this.healthCheck ? renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
-      },
+      certificate: tlsCertificateConfig.tlsCertificate,
+      mode: tlsCertificateConfig.tlsMode.toString(),
     };
   }
-}
 
-function renderHealthCheck(
```

Comment:
For the sake of making the diff read better, let's leave this as a function (and let's also make `renderTls` a function instead of a method).